lib/client.c

```diff
@@ -802,7 +813,7 @@ SF_STATUS STDCALL snowflake_connect(SF_CONNECT *sf) {
         // connection object doesn't exists
         return SF_STATUS_ERROR_CONNECTION_NOT_EXIST;
     }
-    if (sf->token || sf->master_token) {
+    if (!is_token_required(sf) && (sf->token || sf->master_token)) {
```

Comment:
@sfc-gh-ext-simba-jy don't reuse the existing token in SF_CONNECT, that's the session token used in the header of every single request. You can check ODBC implementation as a reference, there are many different tokens in connection class, (token, session token, master token, id token, mfa token).